lib/commands/misc.js

```diff
@@ -0,0 +1,134 @@
+
+    cy.url().then(function (url) {
+        if (url.includes("cas/login") && retryCount < env.GetMaxRetryCount()) {
+            ++retryCount
+            cy.login(username, password, retryCount)
+        }
+    })
+}
+
+
+/**
+ * Logs the defined admin user into the ces. The data for the admin user can be defined in the cypress.json.
+ */
+const loginAdmin = () => {
+    cy.visit("/" + env.GetDoguName())
+    cy.clickWarpMenuCheckboxIfPossible()
+    cy.login(env.GetAdminUsername(), env.GetAdminPassword());
+}
+
+/**
+ * Log the current user out of the cas via back-channel logout.
+ */
+const logout = () => {
+    cy.visit("/cas/logout")
+    //Give cas some time to send the logout requests to the dogus
+    cy.wait(1000)
+}
+
+/**
+ * Handles the warp menu tooltip by clicking the 'do not show again' checkbox on the first time.
+ */
+const clickWarpMenuCheckboxIfPossible = () => {
+    cy.get('div[id="warp-menu-container"]').then(function (container) {
+        let warpContainer = container.children(".warp-menu-column-tooltip")
+        if (warpContainer.length === 1) {
+            cy.get('input[type="checkbox"]').click(true)
+        }
+    })
+}
+
+/**
+ * Returns whether the given user has administrative privileges in the CES.
+ * @param {String} username - The username of the user to check.
+ */
+const isCesAdmin = (username) => {
+    cy.usermgtGetUser(username).then(function (response) {
+        for (var element of response.memberOf) {
+            if (element === env.GetAdminGroup()) {
+                return true
+            }
+        }
+        return false
+    })
+}
+
+/**
+ * Promotes an account to a ces admin account. If the given account is already admin it does nothing.
+ * @param {String} username - The username of the user to promote.
+ */
+const promoteAccountToAdmin = (username) => {
+    cy.isCesAdmin(username).then(function (isAdmin) {
+        if (!isAdmin) {
+            cy.usermgtAddMemberToGroup(env.GetAdminGroup(), username)
+        }
+    })
+}
+
+/**
+ * Demotes an account to a ces default account. If the given account is already a default account it does nothing.
+ * @param {String} username - The username of the user to demote.
+ */
+const demoteAccountToDefault = (username) => {
+    cy.isCesAdmin(username).then(function (isAdmin) {
+        if (isAdmin) {
+            cy.usermgtRemoveMemberFromGroup(env.GetAdminGroup(), username)
+        }
+    })
+}
+
+/**
+ * Wait for a dogu to be healthy. Healthy means it returns a status code less than 400 on
+ * normal get request to the dogu page.
+ * @param {number} currentWaitTimeInMs - recursive parameter - should always be set to 0 when calling the method.
+ * @param {number} timeOutInMs - The maximum time to wait for the dogu to be healthy.
+ * @param {number} waitTimeAfterRequest - The time to wait after one request.
+ */
+const checkDoguHealthLessThanHTTP400 = (currentWaitTimeInMs, timeOutInMs, waitTimeAfterRequest) => {
```

Comment:
The name of the function might be misleading as the 400 in the name implies that its use case are faulty dogus. Maybe sth. like "checkDoguHealty" is sufficient.
Another point is that this function can cause our test to abort if the pipeline is heavily loaded, simply because we have estimated the timeout badly. Can this become a problem?